multicluster/controllers/multicluster/core/remote_cluster.go

```diff
@@ -0,0 +1,353 @@
+	memberAnnounceList := &multiclusterv1alpha1.MemberClusterAnnounceList{}
+	if err := r.List(context.TODO(), memberAnnounceList, client.InNamespace(r.GetNamespace())); err != nil {
+		return err
+	}
+	var localClusterMemberAnnounce multiclusterv1alpha1.MemberClusterAnnounce
+	localClusterMemberAnnounceExist := false
+	if len(memberAnnounceList.Items) != 0 {
+		for _, memberAnnounce := range memberAnnounceList.Items {
+			if memberAnnounce.ClusterID == string(r.remoteClusterManager.GetLocalClusterID()) {
+				localClusterMemberAnnounceExist = true
+				localClusterMemberAnnounce = memberAnnounce
+				break
+			}
+		}
+	}
+	ctx, cancelFunc := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancelFunc()
+	if localClusterMemberAnnounceExist {
+		if r.remoteClusterManager.GetElectedLeaderClusterID() != common.INVALID_CLUSTER_ID {
+			localClusterMemberAnnounce.LeaderClusterID = string(r.remoteClusterManager.GetElectedLeaderClusterID())
+		}
+
+		if localClusterMemberAnnounce.Annotations == nil {
+			localClusterMemberAnnounce.Annotations = make(map[string]string)
+		}
+		localClusterMemberAnnounce.Annotations["touch-ts"] = time.Now().String()
+		// do an update
+		if err := r.Update(ctx, &localClusterMemberAnnounce, &client.UpdateOptions{}); err != nil {
+			klog.Error(err, "Failed to update member announce", "cluster", r.GetClusterID())
+			return err
+		}
+	} else {
+		// create happens first before a leader can be elected, when the create is successful
+		// it marks the connectivity status to leader election can occur
+		// Therefore the first create will not populate the leader cluster id
+		localClusterMemberAnnounce.ClusterID = string(r.remoteClusterManager.GetLocalClusterID())
+		localClusterMemberAnnounce.Name = "member-announce-from-" + string(r.remoteClusterManager.GetLocalClusterID())
+		localClusterMemberAnnounce.Namespace = r.Namespace
+		localClusterMemberAnnounce.ClusterSetID = string(r.ClusterSetID)
+		if err := r.Create(ctx, &localClusterMemberAnnounce, &client.CreateOptions{}); err != nil {
+			klog.Error(err, "Failed to create member announce", "cluster", r.GetClusterID())
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *remoteCluster) updateRemoteClusterStatus(connected bool, err error) {
+	if r.connected == connected {
+		return
+	}
+
+	klog.Infof("Updating remote cluster status", "cluster", r.GetClusterID(), "connected", connected)
+
+	// TODO: Tolerate transient failures so we dont oscillate between connected and disconnected.
+	r.connected = connected
+}
+
+/**
+ * ---------------------------
+ * commonArea Implementation
+ * ---------------------------
+ */
+
+func (r *remoteCluster) GetClusterID() common.ClusterID {
+	return r.ClusterID
+}
+
+func (r *remoteCluster) GetNamespace() string {
+	return r.Namespace
+}
+
+/**
+ * ---------------------------
+ * RemoteCluster Implementation
+ * ---------------------------
+ */
+
+/**
+ * Once connected to the remote cluster, the start method run a timer
```

Comment:
the start method -> Start

run -> runs

Reply:
Done, thanks.